apps/client/src/app/components/transactions-table/transactions-table.component.ts

```diff
@@ -169,4 +227,40 @@ export class TransactionsTableComponent
     this.unsubscribeSubject.next();
     this.unsubscribeSubject.complete();
   }
+
+  private updateFilter() {
+    this.dataSource.filter = this.searchKeywords.join(SEARCH_STRING_SEPARATOR);
+    const lowercaseSearchKeywords = this.searchKeywords.map((keyword) =>
+      keyword.trim().toLowerCase()
+    );
+    this.allFilteredTransactions = TransactionsTableComponent.getSearchableFieldValues(
+      this.transactions
+    ).filter((item) => {
+      return !lowercaseSearchKeywords.includes(item.trim().toLowerCase());
+    });
+    this.filteredTransactions$.next(this.allFilteredTransactions);
+  }
+
+  private static getSearchableFieldValues(
+    transactions: OrderWithAccount[]
+  ): string[] {
+    const fieldValues = new Set<string>();
+    for (const transaction of transactions) {
+      this.getFilterableValues(transaction, fieldValues);
+    }
+
+    return [...fieldValues].sort();
+  }
+
+  private static getFilterableValues(
+    transaction,
+    fieldValues: Set<string> = new Set<string>()
+  ): string[] {
+    fieldValues.add(transaction.currency);
+    fieldValues.add(transaction.symbol);
+    fieldValues.add(transaction.type);
+    fieldValues.add(transaction.Account?.name);
+    fieldValues.add(transaction.Account?.Platform?.name);
```

Comment:
If some of these fields can be undefined or null you have to check and you don't need to add to fieldValues, otherwise it can cause errors in the rest of the code you have written. 

Ex. https://github.com/ghostfolio/ghostfolio/pull/76/files#diff-23494c0c884566b94e8cf55674adf63fe2007424d0dfc8c765347d6691b395a0R239

Reply:
Good catch, thank you. I fixed it.